### PR TITLE
Allow discount code to be applied on tickets and fix discount code model

### LINKED ISSUE #1215: Create Discount code doesn't allow applying it on Tickets
**Describe the bug**

The create discount code route doesn't even fetch the list of Tickets. As a result it doesn't allow applying the discount code to one or more tickets.

**To Reproduce**
1. Go to view event
2. Click on Tickets -> Discount code -> Create Discount code
3. There are no tickets to apply the discount code to.

**Expected behavior**

It should list down the tickets with checkboxes to select one,multiple or none tickets.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by pradeepgangwar**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Currently creation of discount code does not relate to tickets. This PR adds relationship to tickets.

#### Changes proposed in this pull request:
- Add option to select tickets to which a discount code can be applied
- Fix some other minor issues.

Fixes #1215

### CODE REVIEW

**Comment by pradeepgangwar** [comment]
Currently server does not has relation set to tickets for discount code schema. There is a issue opened for the same. https://github.com/fossasia/open-event-server/issues/4905

**Comment by pradeepgangwar** [comment]
@niranjan94 On creating a new record the list does not update itself. We have to refresh the page. I tries several approaches but they were either inefficient or failed. I tired reloading parent's `events.view` model and querying on that but it didn't work.
```
//routes/events/view/tickets/discount-codes/list.js
let eventModel = this.modelFor('events.view').reload();
return eventModel.query('discountCodes', {
      filter       : filterOptions,
      'page[size]' : 10
    });
```
Also on searching many suggested using `findAll()` and `find()` to get live model. But that seems inefficient here as currently we are using parent's model and doing `findAll` for discount-codes we might have to change adapter to query to route `events/id/discount-codes`.
Please help if there is a solution for this. Multiple pages in our app face the similar problem.

**Comment by pradeepgangwar** [comment]
@niranjan94 @ritikamotwani @mayank8318 @srv-twry Please review. :)

File: app/templates/components/ui-table/cell/events/view/tickets/discount-codes/cell-actions.hbs [hunk 1]
```diff
@@ -5,7 +5,7 @@
   {{#ui-popup content=(t 'Deactivate') class='ui icon button' position='left center'}}
     <i class="remove icon"></i>
   {{/ui-popup}}
-  {{#ui-popup content=(t 'Delete') class='ui icon button' position='left center'}}
+  {{#ui-popup content=(t 'Delete') click=(action (confirm (t 'Are you sure you would like to delete this Discount Code?') (action deleteDiscountCode record))) class='ui icon button' position='left center'}}
```
Comment (CosmicCoder96) [inline app/templates/components/ui-table/cell/events/view/tickets/discount-codes/cell-actions.hbs]: @pradeepgangwar The component test is failing, you need to update the test file to accommodate the action you are passing.

**Comment by pradeepgangwar** [comment]
@geekyd @CosmicCoder96 @ritikamotwani @mayank8318 Please review. :)

**Pull request merged**